tmt/steps/report/polarion.py

```diff
@@ -0,0 +1,121 @@
+
+DEFAULT_NAME = 'xunit.xml'
+
+
+class ReportPolarion(tmt.steps.report.ReportPlugin):
+    """
+    Write test results into a xUnit file and upload to Polarion
+    """
+    _methods = [tmt.steps.Method(name='polarion', doc=__doc__, order=50)]
+    _keys = ['file', 'no-upload', 'project-id', 'testrun-title']
+
+    @classmethod
+    def options(cls, how: Optional[str] = None) -> List[tmt.options.ClickOptionDecoratorType]:
+        """ Prepare command line options """
+        options = super().options(how)
+        options[:0] = [
+            click.option(
+                '--file', metavar='FILE', help='Path to the file to store xUnit in'),
+            click.option(
+                '--upload / --no-upload', default=True, show_default=True,
+                help="Whether to upload results to Polarion"),
+            click.option(
+                '--project-id', required=True, help='Use specific Polarion project ID'),
+            click.option(
+                '--testrun-title', help='Use specific TestRun title')
+            ]
+        return options
+
+    def go(self, *args: Any, **kwargs: Any) -> None:
+        """ Go through executed tests and report into Polarion """
+        super().go()
+
+        from tmt.export import get_polarion_ids, import_polarion
+        import_polarion()
+        from tmt.export import PolarionWorkItem
+        assert PolarionWorkItem
+
+        title = self.opt(
+            'testrun_title',
+            self.step.plan.name.rsplit('/', 1)[1] +
+            datetime.datetime.now().strftime("%Y%m%d%H%M%S"))
+        title = title.replace('-', '_')
+        project_id = self.opt('project-id')
+        upload = self.opt('upload')
+
+        junit_suite = make_junit_xml(self)
+        xml_tree = ET.fromstring(junit_suite.to_xml_string([junit_suite]))
+
+        properties = {
+            'polarion-project-id': project_id,
+            'polarion-user-id': PolarionWorkItem._session.user_id,
+            'polarion-testrun-id': title,
+            'polarion-project-span-ids': project_id}
+        testsuites_properties = ET.SubElement(xml_tree, 'properties')
+        for name, value in properties.items():
+            ET.SubElement(testsuites_properties, 'property', attrib={
+                'name': name, 'value': value})
+
+        testsuite = xml_tree.find('testsuite')
+        project_span_ids = xml_tree.find(
+            '*property[@name="polarion-project-span-ids"]')
+
+        for result in self.step.plan.execute.results():
+            if not result.id:
+                raise tmt.utils.ReportError(
+                    f"Test Case {result.name} is not exported to Polarion, "
+                    "please run 'tmt tests export --how polarion' on it")
+            work_item_id, test_project_id = get_polarion_ids(
+                PolarionWorkItem.query(
+                    result.id, fields=['work_item_id', 'project_id']))
+
+            if test_project_id not in project_span_ids.attrib['value']:
```

Comment:
`tmt/steps/report/polarion.py:83: error: Item "None" of "Optional[Element]" has no attribute "attrib"  [union-attr]`

That's correct, `project_span_ids` as a return value of `xml_tree.find()` is indeed `Optional[Element]`, and nothing tests whether it's a valid value before trying to access its attributes. `if project_span_ids is None: raise error` or `assert project_span_ids is not None` would help mypy, with the former giving us better chance to handle the error situation with a more helpful message or a completely different set of actions.